projects/plutus/inc/current_sense.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+// Current sense module for the LTC2484 ADC
+
+#include "ltc_adc.h"
+#include "status.h"
+
+typedef void (*CurrentSenseCallback)(int32_t current, void *context);
+
+typedef struct {
+  int32_t voltage;  // Voltage in microvolts
+  int32_t current;  // Voltage in milliamps
+} CurrentSenseValue;
+
+// User-defined points for two-point calibration
+typedef struct {
+  CurrentSenseValue zero_point;
+  CurrentSenseValue max_point;
+} CurrentSenseCalibrationData;
+
+typedef struct {
+  LtcAdcStorage *adc_storage;
+  CurrentSenseCalibrationData *data;
+  int32_t current_value;
+  volatile uint8_t samples;
+  CurrentSenseCallback callback;
+  void *context;
+} CurrentSenseStorage;
+
+// Initialize the current sense module. Requires CurrentSenseLineData to be calibrated beforehand
+StatusCode current_sense_init(CurrentSenseStorage *storage, CurrentSenseCalibrationData *data,
```

Comment:
Is the ADC supposed to be initialized beforehand or are you initializing it? Please make that clear.

If you're the only consumer of the ADC and you plan on initializing it yourself, you should just allocate the storage in your current sense storage.

Reply:
Fixed